tests/simulator_test.cpp

```diff
@@ -2,82 +2,217 @@
-          manager.collectorRadius() * std::sin(deg2rad(60)));
-
-  ASSERT_EQ(hitData.collisionPoint(), referenceCollisionAtSixty)
-      << "Invalid hit from " << atSixtyDegreesAlongX;
-
-  Ray straightDown(kVZero, core::Vec3(0, 0, -1));
-  ASSERT_FALSE(performHitAtEnergyCollectors(straightDown, &hitData))
-      << "hit registered in: " << hitData.collisionPoint();
-
-  // Ray at the edge of EnergyCollector at: Vec3(0, 0, 2)
-  Ray straightUpAbove(
-      core::Vec3(0, 0,
-                 manager.simulatiorRadius() / 2 + manager.collectorRadius()),
-      core::Vec3(0, 0, 1));
-  ASSERT_FALSE(performHitAtEnergyCollectors(straightUpAbove, &hitData))
-      << "hit registered in: " << hitData.collisionPoint();
-
-  Ray insideEnergyCollectorStraightUp(
-      core::Vec3(manager.simulatiorRadius() / 2, 0, 0), core::Vec3(0, 0, 1));
-  ASSERT_TRUE(
-      performHitAtEnergyCollectors(insideEnergyCollectorStraightUp, &hitData));
-
-  ASSERT_EQ(
-      core::Vec3(manager.simulatiorRadius() / 2, 0, manager.collectorRadius()),
-      hitData.collisionPoint());
-
-  Ray tiltedFromXY(kVZero, Vec3(2 * manager.collectorRadius(),
-                                2 * manager.collectorRadius(), 2));
-  ASSERT_FALSE(performHitAtEnergyCollectors(tiltedFromXY, &hitData));
-
-  Ray straightUpMoved(
-      Vec3(2 * manager.collectorRadius(), 2 * manager.collectorRadius(), 0),
-      Vec3(0, 0, 1));
-  ASSERT_FALSE(performHitAtEnergyCollectors(straightUpMoved, &hitData));
-} */
\ No newline at end of file
+  ASSERT_EQ(performHitCollector(energyCollectors, straightUp, &hitData), true);
+
+  const float collectorPositionRadius = 4;
+  const float collectorAngle = 2 * kPi / (numCollectors - 1);
+  const float refCollectorRadius =
+      collectorPositionRadius * std::sqrt(2 - 2 * std::cos(collectorAngle));
+
+  Vec3 collisionPointStraightUp =
+      Vec3(0, 0, collectorPositionRadius - refCollectorRadius);
+  ASSERT_EQ(hitData.collisionPoint(), collisionPointStraightUp);
+
+  Ray straightDown(kVecZero, -kVecUp);
+  ASSERT_FALSE(performHitCollector(energyCollectors, straightDown, &hitData))
+      << "Collision Point: " << hitData.collisionPoint();
+
+  Ray alongX(kVecZero, kVecX);
+  ASSERT_TRUE(performHitCollector(energyCollectors, alongX, &hitData));
+
+  ASSERT_EQ(alongX.at(collectorPositionRadius - refCollectorRadius),
+            hitData.collisionPoint());
+
+  Ray alongY(kVecZero, kVecY);
+  ASSERT_TRUE(performHitCollector(energyCollectors, alongY, &hitData));
+
+  ASSERT_EQ(alongY.at(collectorPositionRadius - refCollectorRadius),
+            hitData.collisionPoint());
+
+  Ray atSixty(kVecZero, Vec3(std::cos(deg2rad(60)), 0, std::sin(deg2rad(60))));
+  ASSERT_TRUE(performHitCollector(energyCollectors, atSixty, &hitData));
+
+  Ray atSixtyOther(kVecZero,
+                   Vec3(-std::cos(deg2rad(60)), 0, std::sin(deg2rad(60))));
+  ASSERT_TRUE(performHitCollector(energyCollectors, atSixtyOther, &hitData));
```

Comment:
Check position/time?

Reply:
nah, its not that hard. Maybe when numCollector is even its not so obvious, but when I have everything written on papaer its easy :-)

Reply:
i will add collision point check on every hit tommorow ;-)